Kernel/Net/TCPSocket.cpp

```diff
@@ -360,18 +360,22 @@ bool TCPSocket::should_delay_next_ack() const
 
 NetworkOrdered<u16> TCPSocket::compute_tcp_checksum(IPv4Address const& source, IPv4Address const& destination, TCPPacket const& packet, u16 payload_size)
 {
-    struct [[gnu::packed]] PseudoHeader {
-        IPv4Address source;
-        IPv4Address destination;
-        u8 zero;
-        u8 protocol;
-        NetworkOrdered<u16> payload_size;
+    union PseudoHeader {
+        struct [[gnu::packed]] {
+            IPv4Address source;
+            IPv4Address destination;
+            u8 zero;
+            u8 protocol;
+            NetworkOrdered<u16> payload_size;
+        } header;
+        u16 raw[6];
     };
+    static_assert(sizeof(PseudoHeader) == 12);
 
-    PseudoHeader pseudo_header { source, destination, 0, (u8)IPv4Protocol::TCP, packet.header_size() + payload_size };
+    PseudoHeader pseudo_header { .header = { source, destination, 0, (u8)IPv4Protocol::TCP, packet.header_size() + payload_size } };
```

Comment:
Pre-existing, but the addition for the size parameter could trivially overflow a u16 -- especially with the integer promotion happening as a result of operator+. Do we verify that won't happen anywhere further up the call stack?

Reply:
Kind of, maybe.

The only place where we calculate the checksum is when we actually send packages, and the only packages that are sent with a payload are in `protocol_send`. That specific function always limits the payload size to the MTU.

It's not an outright verification, but that MTU limit should place the size of an individual package way below the critical limit for anything that we support as of now.

Reply:
Seems like something we could `VERIFY()` here?